Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -192,84 +157,67 @@ class APIClient {
             task.resume()
         }
     }
-    
+
+    private func refreshToken(completion: @escaping () -> Void) {
+        guard _isRefreshingToken.compareAndSwap(old: false, new: true) else {
+            completion()
+            return
+        }
+
+        // We stop the queue so no more operations are triggered
+        operationQueue.isSuspended = true
+
+        // Increase the amount of consecutive failures
+        _tokenRefreshConsecutiveFailures.mutate { $0 += 1 }
+
+        tokenRefresher { [weak self] in
+            self?.isRefreshingToken = false
+            // We restart the queue now that token refresh is completed
+            self?.operationQueue.isSuspended = false
+            completion()
+        }
+    }
```

Comment:
The expired token error can also come via the web socket so, eventually, we'll need to encapsulate the refreshing logic with all the state info.

It's also weird we don't propagate token refreshing results so we don't know if the token is refreshed without letting the next request go, right? 🤔